### ISSUE TITLE
Get notifications in case of decoding errors

### ISSUE THREAD

**Issue by ksdon** [ZeroCM/zcm#228]
Hi,
we use ZCM for our C++ components in a distributed system. The main application runs on an embedded Linux system sending debugging message via ZCM to a monitoring system.

In order to get some more information about communication errors we want to detect decoding errors. Currently console output is printed. This does not fit our need since we need to react immediately if someone disturbs our system.

It would be nice to have such a notification e.g. by setting user-defined callbacks triggered if decoding errors occur.  

Have you planned to implement something like this?

**Comment by jbendes** [comment]
So what exactly do you mean by decoding errors? I think I might be interpreting your question differently than you intend. A "decoding error" happens at dispatch time - that is when zcm goes to dispatch a message to a typed subscription a decoding errors occurs if the message is not of the type that the subscription is expecting. 

You won't ever get a corrupted message delivered to a subscription because the transport is responsible for ensuring integrity of the data sent and received. If you're looking for a way to get at byte and bit errors at the transport level, I think that's a totally separate but valid question to be asking. 

Can you clarify what exactly you mean here?

**Comment by ksdon** [comment]
Yes, I mean decoding errors during dispatch time. We have several Components using ZCM (embedded and High Level) and we want to know if they are working as expected. For example if we change message types and one of the components is sending still the old one.

**Comment by jbendes** [comment]
Ahhh makes sense. Do you have a recommended way of integrating that functionality into the zcm API? I can brainstorm too and see if I come up with anything good

**Comment by ksdon** [comment]
I could implement a proposal

**Comment by jbendes** [comment]
Sorry just saw I never responded! I think your solution above sounds great. Still interested in pullreq-ing that?

**Comment by erniejunior** [comment]
Yes, but the PR with the test goes first.
